app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,20 @@
+class MoviesController < ApplicationController
+
+  def show
+    @movie = Movie.find(params[:id])
+    @actor_names = @movie.order_by_age
+    @avg_age = @movie.avg_age
+    if params[:search_by_name].present? 
+      threshold = params[:search_by_name].to_s
+      @actors = Actor.filter_by_name(threshold)
+    end
+  end
+
+  def update
```

Comment:
Your implementation here is good but a more RESTful option for this action would be a POST because you are creating an association between 2 tables (i.e. you're creating a joins record by creating a MovieActor). 

One RESTful option would be to create a joins controller called the MovieActors controller, and have the form in the view direct to a #create action there. The create action could have this exact logic